src/attribute.rs

```diff
@@ -16,43 +17,49 @@ pub(crate) enum Attribute<'src> {
 }
 
 impl<'src> Attribute<'src> {
-  pub(crate) fn from_name(name: Name) -> Option<Self> {
-    name.lexeme().parse().ok()
-  }
-
-  pub(crate) fn name(&self) -> &'static str {
-    self.into()
-  }
-
-  pub(crate) fn with_argument(
-    self,
+  pub(crate) fn parse(
     name: Name<'src>,
-    argument: StringLiteral<'src>,
+    maybe_argument: Option<&'src str>,
   ) -> CompileResult<'src, Self> {
-    match self {
-      Self::Confirm(_) => Ok(Self::Confirm(Some(argument))),
-      _ => Err(name.error(CompileErrorKind::UnexpectedAttributeArgument { attribute: self })),
-    }
+    let name_str = name.lexeme();
+    Ok(match (name_str, maybe_argument) {
+      ("group", Some(name)) => Self::Group { name },
+      ("group", None) => {
+        return Err(name.error(CompileErrorKind::MissingAttributeArgument {
+          attribute_name: "group".into(),
+        }))
+      }
+      ("confirm", argument) => Self::Confirm(argument),
+      (other_attribute, None) => other_attribute.parse().map_err(|_| {
+        name.error(CompileErrorKind::UnknownAttribute {
+          attribute: name_str,
+        })
+      })?,
+      (_other_attribute, Some(_)) => {
+        return Err(name.error(CompileErrorKind::UnexpectedAttributeArgument {
+          attribute: name_str,
+        }))
+      }
+    })
   }
 
-  fn argument(&self) -> Option<&StringLiteral> {
-    if let Self::Confirm(prompt) = self {
-      prompt.as_ref()
-    } else {
-      None
-    }
+  pub(crate) fn name(&self) -> &'static str {
+    self.into()
   }
 }
 
 impl<'src> Display for Attribute<'src> {
   fn fmt(&self, f: &mut Formatter) -> Result<(), fmt::Error> {
-    write!(f, "{}", self.name())?;
-
-    if let Some(argument) = self.argument() {
-      write!(f, "({argument})")?;
+    let attr_name = self.name();
+    match self {
+      Self::Confirm(Some(prompt)) => write!(f, "{attr_name}('{prompt}')"),
+      Self::Group { name } => {
+        let use_quotes = name.contains(char::is_whitespace);
+        let mq = if use_quotes { "\"" } else { "" };
+        write!(f, "{attr_name}({mq}{name}{mq})")
```

Comment:
This will not work in general. Single quoted strings, and triple-delimited strings might have content that can't be contained in a single `"`. I think the simplest thing is to punt on accepting bare words, and only accept string literals for the time being.

Reply:
Makes sense. I modified `Group` to take a `StringLiteral`.